drivers/digital-io/max22200/max22200.c

```diff
@@ -0,0 +1,958 @@
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#include <errno.h>
+#include <stdlib.h>
+#include <string.h>
+#include "max22200.h"
+#include "no_os_util.h"
+#include "no_os_alloc.h"
+#include "no_os_delay.h"
+
+/**
+ * @brief Read data from desired register for MAX22200
+ * @param desc - MAX22200 device descriptor.
+ * @param reg - Register value to read data from.
+ * @param val - Value read from the register.
+ * @param one_byte - true - Only the register's LSB will be returned through val
+ * 		     false - Whole 32bit register will be returned through val.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_reg_read(struct max22200_desc *desc, uint32_t reg, uint32_t *val,
+		      bool one_byte)
+{
+	struct no_os_spi_msg xfer = {
+		.tx_buff = desc->buff,
+		.rx_buff = desc->buff,
+		.bytes_number = 1,
+		.cs_change = 1,
+	};
+	int ret;
+
+	desc->buff[0] = no_os_field_prep(MAX22200_RW_MASK, 0) |
+			no_os_field_prep(MAX22200_ADDR_MASK, reg) |
+			no_os_field_prep(MAX22200_ONE_BYTE_MASK, one_byte);
+
+	ret = no_os_gpio_set_value(desc->cmd_desc, NO_OS_GPIO_HIGH);
+	if (ret)
+		return ret;
+
+	ret = no_os_spi_transfer(desc->comm_desc, &xfer, 1);
+	if (ret)
+		return ret;
+
+	if (!one_byte)
+		xfer.bytes_number = MAX22200_FRAME_SIZE;
+
+	ret = no_os_gpio_set_value(desc->cmd_desc, NO_OS_GPIO_LOW);
+	if (ret)
+		return ret;
+
+	ret = no_os_spi_transfer(desc->comm_desc, &xfer, 1);
+	if (ret)
+		return ret;
+
+	if (one_byte) {
+		*val = desc->buff[0];
+		return 0;
+	}
+
+	*val = no_os_get_unaligned_be32(desc->buff);
+
+	return 0;
+}
+
+/**
+ * @brief Write data to desired register for MAX22200
+ * @param desc - MAXX22200 device descriptor.
+ * @param reg - Register value to write data to.
+ * @param val - Value to be written to th register.
+ * @param one_byte - true - Only the register's LSB will be written.
+ * 			    If val is greater than 0xFF and one_byte is true,
+ * 			    function will return negative error code.
+ * 		     false - The whole 32bit register will be written.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_reg_write(struct max22200_desc *desc, uint32_t reg, uint32_t val,
+		       bool one_byte)
+{
+	struct no_os_spi_msg xfer = {
+		.tx_buff = desc->buff,
+		.rx_buff = desc->buff,
+		.bytes_number = 1,
+		.cs_change = 1,
+	};
+	int ret;
+
+	if (one_byte && val > 0xFF)
+		return -EINVAL;
+
+	desc->buff[0] = no_os_field_prep(MAX22200_RW_MASK, 1) |
+			no_os_field_prep(MAX22200_ADDR_MASK, reg) |
+			no_os_field_prep(MAX22200_ONE_BYTE_MASK, one_byte);
+
+	ret = no_os_gpio_set_value(desc->cmd_desc, NO_OS_GPIO_HIGH);
+	if (ret)
+		return ret;
+
+	ret = no_os_spi_transfer(desc->comm_desc, &xfer, 1);
+	if (ret)
+		return ret;
+
+	if (!one_byte) {
```

Comment:
either use brackets for both if and else statement, either you don't for both.

Reply:
Adding brackets to the else statement.